libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -0,0 +1,760 @@
+                                       "*                                                                                      *" +
+                                       "****************************************************************************************" +
+                                       ">> Slack docs: https://api.slack.com/docs/verifying-requests-from-slack";
+
+                throw new Exception(warning + Environment.NewLine + "Required: include a verificationToken or clientSigningSecret to verify incoming Events API webhooks");
+            }
+
+            _api = new SlackTaskClient(options.BotToken);
+            LoginWithSlackAsync(default).Wait();
+        }
+
+        /// <summary>
+        /// Gets the SlackAdapterOptions.
+        /// </summary>
+        /// <value>
+        /// An object containing API credentials, a webhook verification token and other options.
+        /// </value>
+        public SlackAdapterOptions Options { get; private set; }
+
+        /// <summary>
+        /// Gets the user identity.
+        /// </summary>
+        /// <value>
+        /// A string containing the user identity.
+        /// </value>
+        public string Identity { get; private set; }
+
+        /// <summary>
+        /// Wraps Slack API's AddReactionAsync method.
+        /// </summary>
+        /// <param name="name">The optional name.</param>
+        /// <param name="channel">The optional channel.</param>
+        /// <param name="timestamp">The optional timestamp.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>The <see cref="ReactionAddedResponse"/> representing the response to the reaction added.</returns>
+        public virtual async Task<ReactionAddedResponse> AddReactionAsync(string name = null, string channel = null, string timestamp = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await _api.AddReactionAsync(name, channel, timestamp).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's APIRequestWithTokenAsync method.
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <param name="postParameters">The parameters to the POST request.</param>
+        /// <typeparam name="TResult">The generic type for the return. Must be of type Response.</typeparam>
+        /// <returns>A <see cref="Task"/> of type T representing the asynchronous operation.</returns>
+        public virtual async Task<TResult> APIRequestWithTokenAsync<TResult>(CancellationToken cancellationToken, params Tuple<string, string>[] postParameters)
+             where TResult : Response
+        {
+            return (postParameters != null) ?
+                await _api.APIRequestWithTokenAsync<TResult>().ConfigureAwait(false) :
```

Comment:
Looks like this conditional statement should be reversed.  In this case await **_api.APIRequestWithTokenAsync<TResult>(postParameters).ConfigureAwait(false);** will only be called when postParameters is null, which doesn't make sense.

Reply:
It's true. We fixed it. Thanks!